File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/model/ValidationDemoDto.java
package com.itangcent.springboot.demo.common.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

/**
 * Validation Demo DTO
 * Used to demonstrate the usage of JSR-303 validation annotations
 * Contains basic non-null validation examples
 */
public class ValidationDemoDto {
    /**
     * ID field
     * Cannot be null
     */
    @NotNull
    private Long id;

    /**
     * Name field
     * Cannot be empty string or null
     */
    @NotBlank
    private String name;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
} 

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/model/ValidationDemoDto.kt
package com.itangcent.common.model

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

/**
 * 数据验证演示DTO
 * 用于演示JSR-303验证注解
 * 包含基本的非空验证示例
 */
data class ValidationDemoDto(
    /**
     * ID字段
     * 不能为空
     */
    @NotNull
    var id: Long? = null,

    /**
     * 名称字段
     * 不能为空字符串或null
     */
    @NotBlank
    var name: String? = null
)